Plugin~/WebRTCPluginTest/ContextTest.cpp

```diff
@@ -0,0 +1,70 @@
+﻿#include "pch.h"
+#include "GraphicsDeviceTestBase.h"
+#include "../WebRTCPlugin/GraphicsDevice/GraphicsDevice.h"
+#include "../WebRTCPlugin/GraphicsDevice/ITexture2D.h"
+#include "../WebRTCPlugin/Codec/EncoderFactory.h"
+#include "../WebRTCPlugin/Codec/IEncoder.h"
+#include "../WebRTCPlugin/Context.h"
+
+using namespace WebRTC;
+
+class ContextTest : public GraphicsDeviceTestBase
+{
+protected:
+    IEncoder* encoder_ = nullptr;
+//    IGraphicsDevice* device_ = nullptr;
+    const int width = 256;
+    const int height = 256;
+    std::unique_ptr<Context> context;
+
+    void SetUp() override {
+        GraphicsDeviceTestBase::SetUp();
+        EXPECT_NE(nullptr, m_device);
+
+        EncoderFactory::GetInstance().Init(width, height, m_device);
+        encoder_ = EncoderFactory::GetInstance().GetEncoder();
+        EXPECT_NE(nullptr, encoder_);
+
+        context = std::make_unique<Context>();
+    }
+    void TearDown() override {
+        EncoderFactory::GetInstance().Shutdown();
+        GraphicsDeviceTestBase::TearDown();
+    }
+};
+TEST_P(ContextTest, InitializeAndFinalizeEncoder) {
+    context->InitializeEncoder(m_device);
+    context->FinalizerEncoder();
```

Comment:
r in the middle is probably not necessary ? FinalizeEncoder ?

Reply:
fixed